woodwork/table_accessor.py

```diff
@@ -80,74 +83,140 @@ def init(self,
                 added to columns based on the specified logical type for the column.
                 If a single boolean is supplied, will apply the same use_standard_tags value to all columns.
                 A dictionary can be used to specify ``use_standard_tags`` values for individual columns.
-                Unspecified columns will use the default value. Defaults to True.
+                Unspecified columns will use the default value of True.
             column_descriptions (dict[str -> str], optional): Dictionary mapping column names to column descriptions.
             column_origins (str, dict[str -> str], optional): Origin of each column. If a string is supplied, it is
                 used as the origin for all columns. A dictionary can be used to set origins for individual columns.
+            validate (bool, optional): Whether parameter and data validation should occur. Defaults to True. Warning:
+                Should be set to False only when parameters and data are known to be valid.
+                Any errors resulting from skipping validation with invalid inputs may not be easily understood.
+        """
+        self.init_with_partial_schema(**kwargs)
+
+    def init_with_full_schema(self, schema: TableSchema, validate: bool = True, **kwargs) -> None:
+        """Initializes Woodwork typing information for a DataFrame with a complete schema.
+
+        Args:
+            schema (Woodwork.TableSchema): Typing information to use for the DataFrame instead of performing inference. Note that any changes made to the
+                schema object after initialization will propagate to the DataFrame. Similarly, to avoid unintended typing information changes, the same schema
+                object should not be shared between DataFrames.
+            validate (bool, optional): Whether parameter and data validation should occur. Defaults to True. Warning: Should be set to False only when
+                parameters and data are known to be valid. Any errors resulting from skipping validation with invalid inputs may not be easily understood.
+        """
+        if validate:
+            _check_schema(self._dataframe, schema)
+            _check_unique_column_names(self._dataframe)
+        self._schema = schema
+
+        extra_params = [key for key in kwargs]
+        if extra_params:
+            warnings.warn("A schema was provided and the following parameters were ignored: " + ", ".join(extra_params), ParametersIgnoredWarning)
+
+    def init_with_partial_schema(self,
+                                 schema: Optional[TableSchema] = None,
+                                 index: Optional[str] = None,
+                                 time_index: Optional[str] = None,
+                                 logical_types: Optional[Dict[ColumnName, Union[str, LogicalType, None]]] = None,
```

Comment:
By doing `from __future__ import annotations` at the top as was mentioned in a previous review comment, you can use `str | LogicalType | None` instead of `Union[str, LogicalType, None]`.  We should probably adopt that convention for all union type annotations.